### PR TITLE
SQL: Replace seconds with days as seed for random generation

### PR DESCRIPTION
**Pull request by bpintea**
Since the same expression evaluates the current second in two different
engines (H2's and ours), it can happen that these will yield different
results. So replace relying on extracting current second with current
day. This could also yield different results, but should be
significantly less likely.

### CODE REVIEW

File: x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec [hunk 1]
```diff
@@ -13,61 +13,61 @@
...
 mod
-SELECT 18446744073709551615 % ROUND(RAND(SECOND(CURRENT_TIMESTAMP())) * 10000) AS x;
+SELECT 18446744073709551615 % ROUND(RAND(DAY(CURRENT_TIMESTAMP())) * 10000) AS x;
 
 abs
-SELECT ABS(9223372036854775808 + ROUND(RAND(SECOND(CURRENT_TIMESTAMP())) * 10000)) AS x;
+SELECT ABS(9223372036854775808 + ROUND(RAND(DAY(CURRENT_TIMESTAMP())) * 10000)) AS x;
 // ceil
-// SELECT CEIL(9223372036854775808 + RAND(SECOND(CURRENT_TIMESTAMP())) * 10000) AS x;
+// SELECT CEIL(9223372036854775808 + RAND(DAY(CURRENT_TIMESTAMP())) * 10000) AS x;
```
Comment (Luegg) [inline x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec:29]: not really related to this PR but are the commented out on purpose?

**Review by Luegg — APPROVED** [review]
lgtm, did this already trigger any CI failures?

**Review by luigidellaquila — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec:29]: There's this comment in the header:
```
ES's CEIL & FLOOR don't work with unsigned_longs (TODO) + they can return longs, while H2's always returns doubles
```

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/sql/qa/server/src/main/resources/unsigned-long.sql-spec:29]: > did this already trigger any CI failures?

forgot to answer: yes, it did a couple.

**Pull request merged**